lib/proxy/validations.rb

```diff
@@ -11,20 +17,42 @@ def valid_mac?(mac)
     false
   end
 
+  def valid_ip?(ip)
+    !!IPAddr.new(ip) rescue false
+  end
+
+  def valid_ip4?(ip)
+    !!(ip =~ Resolv::IPv4::Regex)
```

Comment:
In modern Ruby (I think 2.5+) you can use
```suggestion
    Resolv::IPv4::Regex.match?(ip)
```